src/modules/site/base/views/mapping.py

```diff
@@ -21,44 +29,75 @@ def mapping():
   """
       This is the mapping submission page.
   """
-  title = 'Perform Mapping'
-  jwt_csrf_token = (get_jwt() or {}).get("csrf")
-  form = FileUploadForm()
-  # TODO: change
-  nemascan_container_url = 'https://github.com/AndersenLab/dockerfile/tree/nemarun/nemarun'
-  nemascan_github_url = 'https://github.com/AndersenLab/NemaScan'
-  return render_template('tools/mapping/mapping.html', **locals())
+  return render_template('tools/mapping/mapping.html', **{
+
+    # Page info
+    'title': 'Perform Mapping',
+
+    # Form info
+    'jwt_csrf_token': (get_jwt() or {}).get("csrf"),
+    'form': FileUploadForm(),
+
+    # TODO: change
+    'nemascan_container_url': 'https://github.com/AndersenLab/dockerfile/tree/nemarun/nemarun',
+    'nemascan_github_url':    'https://github.com/AndersenLab/NemaScan',
+
+    # Species list
+    'species_list': SPECIES_LIST,
+    'species_fields': [
+      'name', 'short_name', 'project_num', 'wb_ver', 'latest_release',
+    ],
+  })
 
 
 @mapping_bp.route('/mapping/upload', methods = ['POST'])
 @jwt_required()
 def submit_mapping_request():
   form = FileUploadForm(request.form)
   user = get_current_user()
+
+  # Validate form
   if not form.validate_on_submit():
     flash("You must include a description of your data and a TSV file to upload", "error")
     return redirect(url_for('mapping.mapping'))
-  
-  label = bleach.clean(request.form.get('label'))
-  props = {'label': label, 
-          'username': user.name,
-          'email': user.email,
-          'file': request.files['file']}          
+
+  # Read fields from form
+  label   = bleach.clean(request.form.get('label'))
+  species = bleach.clean(request.form.get('species'))
+
+  # Save uploaded file to server temporarily with unique generated name
+  # TODO: Is there a better way to generate this name? E.g. using a Tempfile?
+  local_path = os.path.join(uploads_dir, secure_filename(f'{ unique_id() }.tsv'))
+  request.files['file'].save(local_path)
+
+  # Package submission data together into dict
+  data = {
+    'label':    label,
+    'species':  species,
+    'filepath': local_path,
+  }
+
   try:
-    logger.warn("Check Duplicates is DISABLED !!!")
-    m = create_new_mapping(**props, check_duplicates=True)
+    # logger.warn("Check Duplicates is DISABLED !!!")
+    m = create_new_mapping(user, data, no_cache=False)

```

Comment:
careful to cleanup -- we shouldn't return from this function without cleaning up the uploads_dir folder (delete the file at `local_path`) on all the paths for this function on exit.

Reply:
Good catch -- I was performing this inside the `create_new_mapping` function, but A) I think it makes more sense to move up to the view function, and B) I haven't added this for heritability yet.

https://github.com/AndersenLab/CAENDR/blob/feature/multiple-species-etl-ops/src/pkg/caendr/caendr/services/nemascan_mapping.py#L63
```
finally:
    try:
      os.remove(data['filepath'])
    except FileNotFoundError:
      pass
```

Reply:
Fixed in 9f68c1cd0eafd697e77e8098cf4ded68852973ab

Reply:
Excellent!